### PR TITLE
Sending data items to bundling services

### PR DESCRIPTION
**Pull request by janekolszak**
- Sender: Copy Bundler as a starting point
- Sender: SQL migrations
- Sender: Monitoring
- Sender: Model definitions
- Sender: Configuration options
- Sender: Base implementation
- Sender: Command and other small fixes
- Sender: Fix configuration

### CODE REVIEW

File: src/send/sender.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+// Receives bundle items from the input channel and sends them to bundlr
+func NewSender(config *config.Config, db *gorm.DB) (self *Sender) {
+	var err error
+
+	self = new(Sender)
+	self.db = db
+
+	// Seed random generator
+	var b [8]byte
```
Comment (szynwelski) [inline src/send/sender.go:43]: I would separate the function for the random generator.

File: src/send/sender.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+			bundleItem, err := self.parse(item)
+			if err != nil {
+				self.Log.WithError(err).
+					WithField("data_item_id", item.DataItemID).
+					Error("Failed parse and validate data item")
+				item.State = model.BundleStateMalformed
+				goto end
+			}
+
+			// Send the bundle item to bundlr
```
Comment (szynwelski) [inline src/send/sender.go:114]: maybe we should rename all bundlr to iris?

File: src/send/sender.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+				return
+			}
+
+			// Update state
+			item.State = model.BundleStateUploaded
+
+			// Update stats
+			self.monitor.GetReport().Sender.State.BundlrSuccess.Inc()
+
+			if item.State == model.BundleStateUploading {
```
Comment (szynwelski) [inline src/send/sender.go:162]: Will this condition ever be met?

File: src/send/controller.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	// SQL database
+	db, err := model.NewConnection(self.Ctx, config, "sender")
+	if err != nil {
+		return
+	}
+
+	// Arweave client
+	arweaveClient := arweave.NewClient(self.Ctx, config)
+
+	// TODO: tutaj musi byc fallback  na arweave.net jak gw nie odpowiada
```
[the remark this answers is not in the document]
Comment (janekolszak) [reply src/send/controller.go:45]: Removed, this problem never occurred

File: src/send/sender.go (hunk 1 above)
Comment (janekolszak) [reply src/send/sender.go:43]: I have no idea what it was for, removed

File: src/send/sender.go (hunk 2 above)
Comment (janekolszak) [reply src/send/sender.go:114]: Done in sender

File: src/send/sender.go (hunk 3 above)
Comment (janekolszak) [reply src/send/sender.go:162]: ups

**Pull request merged**